scripts/util/HelperNode.lua

```diff
@@ -0,0 +1,110 @@
+--[[
+This file is part of Courseplay (https://github.com/Courseplay/FS22_Courseplay)
+Copyright (C) 2024 Courseplay Dev Team
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+]]--
+
+-- A helper node is a temporary node to help with relative vehicle position calculations.
+-- It wraps the creation of the Giant's engine node.
+-- NOTE: it must be destroyed explicitly to avoid node leak.
+
+---@class HelperNode
```

Comment:
This is a fantastic idea :)
Maybe you got the motivation to add missing commanly used functions, like worldToLocal(), localDirectionToWorld().